### ISSUE TITLE
RFC - Ignore specific query strings

### ISSUE THREAD

**Issue by lenvanessen** [putyourlightson/craft-blitz#177]
Hi There,

First of all, awesome work on Blitz.
We know there is a setting called "Do not cache URLs with query string". But for us, this doesn't offer enough control. I'll explain why this is:

We want to exclude certain query strings, like source or gclid as they are appended quite often.
However, we do want other query strings to be cached as pages, like "?sort=asc" or other filters that are appended to the url.

Would it be possible, to somewhere in the future, include a feature to blacklist certain query strings from being cached? We can ignore these from the exclude uri patterns, but that's not what we're looking for. We don't want an uncached page to be loaded, we just want it to default back to the cache from this page without the url param in some cases.

**Comment by bencroker** [comment]
> Would it be possible, to somewhere in the future, include a feature to blacklist certain query strings from being cached? We can ignore these from the exclude uri patterns, but that's not what we're looking for. We don't want an uncached page to be loaded, we just want it to default back to the cache from this page without the url param in some cases.

So am I understanding correctly that you want to have the `Query String Caching` setting be one value for some query strings and another value for others? Wouldn't your use-case work with the value set to `Cache URLs with query strings as unique pages`?

**Comment by lenvanessen** [comment]
Hi Ben,

On certain query strings, I want it to load the regular cached page without the query string. I'll give you an example to clarify a bit further.

*This is the page that is cached*
https://awesomefastblitzsite.com/products

*A visitor coming from a adwords campaign lands on this url*
https://awesomefastblitzsite.com/products?gclid=1243241424142

But another visitor coming from the same ad lands on the url
https://awesomefastblitzsite.com/products?gclid=9999999999

Since the ID is always unique, but the content isn't, in this case we'd like the setting to be "Cache URLs with query strings as the same page". 

But in the following example, we don't want it to treat it as the same page, as the content is actually different
https://awesomefastblitzsite.com/products?sort=asc

or 
https://awesomefastblitzsite.com/products?brand=nike

So to recap, to prevent issues with our sorting/filtering we've set the setting to "Cache URLs with query strings as unique pages". But this means every user coming from adwords will have to load an uncached version of the site, given that their gclid is unique.

**Comment by bencroker** [comment]
Ok thank you for the clarification. I'll look into adding a config setting to exclude specific query string parameters and will give you an update next week.

**Comment by bencroker** [comment]
Added an `excludedQueryStringParams` config setting in version 3.5.0.
https://github.com/putyourlightson/craft-blitz/blob/3.5.0/src/config.php#L126-L127
